ICAFIX/scripts/RecleanClassify.sh

```diff
@@ -0,0 +1,278 @@
+    count=$(echo "$ModelToUse" | awk -F'@' '{print NF}')
+    VoteThresh=$((count - 1))
+fi
+
+IFS='@' read -a fMRINamesArray <<<"$fMRINames"
+
+tempfiles_create fMRIList_XXXXXX.txt fMRIListName
+
+# check SR or MR FIX
+if [[ "$MRFixConcatName" != "" ]] ; then
+    fMRINamesToUse=("$MRFixConcatName")
+else
+    fMRINamesToUse=("${fMRINamesArray[@]}")
+fi
+# check if FIX features are generated (csv with 181 features)
+for fMRIName in "${fMRINamesToUse[@]}" ; do
+    echo "${fMRIName}" >> "$fMRIListName"
+    FixFeaturePath="${StudyFolder}/${Subject}/MNINonLinear/Results/${fMRIName}/${fMRIName}_hp${HighPass}.ica/fix/features.csv"
+    if [[ ! -e "$FixFeaturePath" ]]; then
+        log_Err_Abort "$FixFeaturePath is not doesn't exist, make sure ICA+FIX is applied to this subject: ${Subject}, fMRI run: ${fMRIName}"
+    fi
+done
+
+# Preprocess step 1: Merge and dropout
+Caret7_Command="wb_command"
+
+#TODO: enable multiple resolutions like macaque (1.25mm resolution) or 7T (1.6mm resolution)
+FinalfMRIResolution="2"
+SmoothingFWHM="2"
+BrainOrdinatesResolution="2"
+Flag="FALSE" #FALSE=Don't fix zeros, TRUE=Fix zeros
+DeleteIntermediates="TRUE" #TRUE/FALSE
+
+CorticalLUT="$HCPPIPEDIR/global/config/FreeSurferCorticalLabelTableLut.txt"
+SubCorticalLUT="$HCPPIPEDIR/global/config/FreeSurferSubcorticalLabelTableLut.txt"
+
+if [ ! ${MRFixConcatName} = "" ] ; then
+    DropOutSubSTRING=""
+    for fMRIName in "${fMRINamesArray[@]}" ; do
+        if [ -e ${StudyFolder}/${Subject}/MNINonLinear/Results/${fMRIName}/${fMRIName}_dropouts.nii.gz ] ; then
+        DropOutSubSTRING=`echo "${DropOutSubSTRING}${StudyFolder}/${Subject}/MNINonLinear/Results/${fMRIName}/${fMRIName}_dropouts.nii.gz "`
+        fi
+    done
+    if [ ! -z "${DropOutSubSTRING}" ] ; then
+        fslmerge -t ${StudyFolder}/${Subject}/MNINonLinear/Results/${MRFixConcatName}/${MRFixConcatName}_dropouts.nii.gz ${DropOutSubSTRING}
+        fslmaths ${StudyFolder}/${Subject}/MNINonLinear/Results/${MRFixConcatName}/${MRFixConcatName}_dropouts.nii.gz -Tmean ${StudyFolder}/${Subject}/MNINonLinear/Results/${MRFixConcatName}/${MRFixConcatName}_dropouts.nii.gz
+        WorkingDirectory="/tmp/RecleanClassify_MR_${Subject}_${MRFixConcatName}"
+        mkdir -p ${WorkingDirectory}
+        cp ${StudyFolder}/${Subject}/MNINonLinear/Results/${MRFixConcatName}/${MRFixConcatName}_dropouts.nii.gz ${WorkingDirectory}/${MRFixConcatName}_dropouts.nii.gz
+        gunzip -f ${WorkingDirectory}/${MRFixConcatName}_dropouts.nii.gz
+        "$HCPPIPEDIR"/ICAFIX/scripts/MapVolumeToCIFTI.sh ${StudyFolder} ${Subject} ${MRFixConcatName} ${CorticalLUT} ${SubCorticalLUT} ${Caret7_Command} ${LowResMesh} ${RegName} ${SmoothingFWHM} ${FinalfMRIResolution} ${BrainOrdinatesResolution} ${WorkingDirectory}/${MRFixConcatName}_dropouts.nii ${StudyFolder}/${Subject}/MNINonLinear/Results/${MRFixConcatName}/${MRFixConcatName}_dropouts.dscalar.nii ${MRFixConcatName}_dropouts ${Flag} ${DeleteIntermediates} nii ${WorkingDirectory}
```

Comment:
```suggestion
        "$HCPPIPEDIR"/ICAFIX/scripts/MapVolumeToCIFTI.sh ${StudyFolder} ${Subject} ${MRFixConcatName} ${CorticalLUT} ${SubCorticalLUT} ${Caret7_Command} ${LowResMesh} ${RegName} ${SmoothingFWHM} ${FinalfMRIResolution} ${BrainOrdinatesResolution} ${WorkingDirectory}/${MRFixConcatName}_dropouts.nii.gz ${StudyFolder}/${Subject}/MNINonLinear/Results/${MRFixConcatName}/${MRFixConcatName}_dropouts.dscalar.nii ${MRFixConcatName}_dropouts ${Flag} ${DeleteIntermediates} nii.gz ${WorkingDirectory}
```
I would hope this works, uncompressed volume files are something we try pretty hard to avoid.  Is there some other tool that needs them to already be uncompressed?  read_avw does its own gunzip, and I would expect nibabel not to have a problem with it...

Same for the gunzip and MapVolume a few lines down.